### PR TITLE
feat: Adding Execution details page

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
This PR added Execution page, displaying execution details including step logs and a execution stage tree view.

In summary:

- Fetches and renders stage, step and overall execution details, including step logs.
- Added a util that parse an `execution` to render an execution status tree format.
- Removed hardcoded avatar url, replaced with user name initials.
- Removed `moment` package from `packages/canary`.
- Cleaned `treeview` component from `packages/canary` - Handle more status types and fixes formatted timestamp.

image

image

### CODE REVIEW

File: apps/gitness/package.json [hunk 1]
```diff
@@ -22,25 +22,26 @@
     "@harnessio/playground": "workspace:*",
     "@harnessio/unified-pipeline": "workspace:*",
     "@harnessio/yaml-editor": "workspace:*",
+    "@harnessio/icons-noir": "workspace:*",
```
Comment (cjlee01) [inline apps/gitness/package.json:25]: how come icon-noir was added again?

File: packages/playground/src/components/execution/execution-details.tsx [hunk 2]
```diff
@@ -26,7 +27,7 @@ export const ExecutionDetails: React.FC<ExecutionProps> = (): React.ReactElement
     <Layout.Horizontal className="px-8">
       {/* Hardcoded height added temporarily */}
```
Comment (cjlee01) [inline packages/playground/src/components/execution/execution-details.tsx:28]: comment needed?

Comment (vardanbansal-harness) [reply packages/playground/src/components/execution/execution-details.tsx]: not needed, removed it

File: apps/gitness/package.json (hunk 1 above)
Comment (vardanbansal-harness) [reply apps/gitness/package.json]: not needed, removed it

File: packages/playground/src/components/execution/execution-details.tsx [hunk 3]
```diff
@@ -22,32 +23,29 @@ export const ExecutionDetails: React.FC<ExecutionProps> = (): React.ReactElement
...
   const stages = execution.stages
   if (!stages || !stages.length) return <Text>No stages found</Text>
+  const stageIdx = 0
+  const stepIdx = 0
   return (
     <Layout.Horizontal className="px-8">
-      {/* Hardcoded height added temporarily */}
       <div className="w-2/3">
-        <StageExecution stage={stages[0]} />
+        <StageExecution stage={stages[stageIdx]} selectedStepIdx={stepIdx} logs={mockStepLogs[stepIdx]} />
```
Comment (cjlee01) [inline packages/playground/src/components/execution/execution-details.tsx:31]: is this still mockdata for steplogs?

File: apps/gitness/src/pages/execution/execution-details.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    stage_number: String(stageNum),
+    step_number: String(stepNum)
+  })
+
+  useEffect(() => {
+    if (execution?.stages && execution.stages.length > 0) {
+      const stageIdx = stageNum > 0 ? stageNum - 1 : 0
+      setStage(execution.stages[stageIdx])
+    }
+  }, [execution?.stages, stageNum])
```
Comment (cjlee01) [inline apps/gitness/src/pages/execution/execution-details.tsx:47]: could we use a useMemo for less rerenders?

**Review by cjlee01 — APPROVED** [review]

File: apps/gitness/src/pages/execution/execution-details.tsx (hunk 4 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/execution/execution-details.tsx:47]: Sure, will fix in a follow up PR.

**Pull request merged**